src/huggingface_hub/keras_mixin.py

```diff
@@ -0,0 +1,103 @@
+import logging
+import os
+from pathlib import Path
+
+from huggingface_hub import ModelHubMixin, hf_hub_download
+
+
+logger = logging.getLogger(__name__)
+
+
+class KerasModelHubMixin(ModelHubMixin):
+
+    _CONFIG_NAME = "config.json"
+    _WEIGHTS_NAME = "tf_model.h5"
+
+    def __init__(self, *args, **kwargs):
+        """
+        Mix this class with your keras-model class for ease process of saving & loading from huggingface-hub
+
+        Example::
+
+            >>> from huggingface_hub import KerasModelHubMixin
+
+            >>> class MyModel(tf.keras.Model, KerasModelHubMixin):
+            ...    def __init__(self, **kwargs):
+            ...        super().__init__()
+            ...        self.config = kwargs.pop("config", None)
+            ...        self.dummy_inputs = ...
+            ...        self.layer = ...
+            ...    def call(self, ...)
+            ...        return ...
+
+            >>> model = MyModel()
+            >>> model.save_pretrained("mymodel", push_to_hub=False) # Saving model weights in the directory
+            >>> model.push_to_hub("mymodel", "model-1") # Pushing model-weights to hf-hub
+
+            >>> # Downloading weights from hf-hub & model will be initialized from those weights
+            >>> model = MyModel.from_pretrained("username/mymodel@main")
+        """
+
+    def _save_pretrained(self, save_directory, dummy_inputs=None, **kwargs):
+
+        dummy_inputs = (
+            dummy_inputs
+            if dummy_inputs is not None
+            else getattr(self, "dummy_inputs", None)
+        )
+
+        if dummy_inputs is None:
+            raise RuntimeError(
+                "You must either provide dummy inputs or have them assigned as an attribute of this model"
+            )
```

Comment:
From a user perspective I'm not sure `dummy_inputs` is very clear. Should we explain it in a docstring?

Reply:
Can do!

Reply:
Added a docstring in `__init__`, since I think that makes the most sense (as it'll be visible to user).